addons/point_of_sale/tools/posbox/overwrite_after_init/etc/nginx/sites-enabled/default

```diff
@@ -38,6 +38,13 @@ server {
         autoindex on;
     }
 
+    # Serve static files
+    location ~ ^/[^/]+/static/.+$ {
+        root /home/pi/;
+        try_files /odoo/addons$uri /;
+        expires 24h;
+    }
+
```

Comment:
I get the point of this line, but why is it needed ? we already use other static files like `/web/static/lib/jquery/jquery.js`, why are they able to get served but not the svg file ?

Reply:
I don't want to put this SVG file in the web module because it is used by hw_posbox_homepage. The problem is that static files are only accessible via the Odoo server under certain conditions. And in particular that the module is installable. Except that hw_posbox_homepage is not installable. So the module's static folder must be accessible via nginx.

https://github.com/odoo/odoo/blob/17.0/odoo/http.py#L2063

Reply:
the lines you attached state that it should be installable OR be an asset `and (manifest['installable'] or manifest['assets'])`, isn't that exactly what you are looking for ?

Reply:
@Julien00859 better with nginx or with assets ?
what is best?

Reply:
The network architecture of nginx makes it very efficient to do IO, like to deliver static files. It also isn't limited in term of how many concurrent requests it can serves while Odoo is limited to its number of workers. So in all cases it is better to let nginx serve static content and to only use Odoo as a fallback.